src/waypoint.jsx

```diff
@@ -188,11 +192,17 @@ export default class Waypoint extends React.Component {
         currentPosition: POSITIONS.inside,
         previousPosition,
         event,
+        waypointTop: bounds.waypointTop,
+        viewportTop: bounds.viewportTop,
+        viewportBottom: bounds.viewportBottom,
       });
       this.props.onLeave.call(this, {
         currentPosition,
         previousPosition: POSITIONS.inside,
         event,
+        waypointTop: bounds.waypointTop,
+        viewportTop: bounds.viewportTop,
+        viewportBottom: bounds.viewportBottom,
```

Comment:
I could, but I would need to add a few babel plugins. If you would prefer that syntax I can do so.
https://babeljs.io/docs/plugins/transform-object-rest-spread/